### DISCUSSION TITLE
"unknown error" while setting up (logging in)

### DISCUSSION THREAD

**Discussion by BenoitDefraigne** [Hyundai-Kia-Connect/kia_uvo#750] [Q&A]
Hello,

I have changed my password on the Kia Connect App, so as indicated on an earlier post I removed the integration from HA and tried to reinstall it. When prompted for login and password I fill it correctly with the new password but I get "! Unexpected error" and all fiels back to blank.

in the logviewer I find the following lines : 
 File "/usr/local/lib/python3.11/site-packages/hyundai_kia_connect_api/KiaUvoApiEU.py", line 1198, in _get_authorization_code_with_form
    login_form_action_url = soup.find("form")["action"].replace("&", "&")
                            ~~~~~~~~~~~~~~~~~^^^^^^^^^^
TypeError: 'NoneType' object is not subscriptable

Can someone help here?

Thank you!

**Comment by cdnninja** [comment]
What region and do you have an & in your password?

**Reply by BenoitDefraigne** [reply]
I am in EUR and do not have an & in my password...

**Comment by BenoitDefraigne** [comment]
Issue solved spontaneously : it seems that due to the password change and HA persistently trying to connect with the old password, the API firewall blocks the account and originating IP address.  I got myself a "blocked by Firewall" error page when trying to connect straight to KIA Connect... until the firewall rule measure was released (+- 48h after last attempt). All working fine now.
